search/query_language_whitebox_test.go

```diff
@@ -270,7 +270,105 @@ func TestParseMap(t *testing.T) {
 		assert.Equal(t, expectedQuery, actualQuery)
 	})
 
+	t.Run(OPTS, func(t *testing.T) {
+		t.Parallel()
+		// given
+		input := fmt.Sprintf(`{"%s": [ {"key1": "value1"}, {"key2": "value2"}]}`, OPTS)
+		// Parsing/Unmarshalling JSON encoding/json
+		fm := map[string]interface{}{}
+		err := json.Unmarshal([]byte(input), &fm)
+		require.Nil(t, err)
+		// when
+		actualQuery := Query{}
+		parseMap(fm, &actualQuery)
+		// then
+		v1 := "value1"
+		v2 := "value2"
+		expectedQuery := Query{Name: OPTS, Children: []Query{
+			{Name: "key1", Value: &v1},
+			{Name: "key2", Value: &v2}},
+		}
```

Comment:
Why not store the options like so? Why implement them using the filter logic?

```go
Query{
	Name: AND, 
        Children: []Query{
		{Name: "SpaceID", Value: &spaceID},
		{Name: "Type", Value: &witID},
	}
	Options: map[string]interface{}{
		"orderby": "system.updated_at",
		"orderdir": "asc", 
		// ... more options
	}
}
```

Reply:
It was easy to reuse existing code. But I can see that going for separate attribute looks better.

Reply:
``` I don't consider the options a query parameter but an addition to a query.``` @kwk Can you please explain this bit? How do you define the scope of query parameter vs query addition?

Reply:
@DhritiShikhar the wording might be odd I know. But consider how we build an expression tree from a given JSON string. That tree should not change, so that the query execution also doesn't change. The only difference should be that we also read options from the top level object in the the JSON string. @baijum 's implementation puts the options inside the expression tree.

Reply:
@kwk I understand your point now. Thanks for the explanation.